### PR TITLE
feat(shield): enable kubernetes-metadata by default and decrease delegated nodes number to 0

### PR DESCRIPTION
**Pull request by AlbertoBarba**
## What this PR does / why we need it:

Enable kubernetes-metadata by default and decrease delegated nodes number to 0

## Checklist

- [x] Title of the PR starts with type and scope, (e.g. `feat(agent,node-analyzer,sysdig-deploy):`)
- [ ] Chart Version bumped for the respective charts
- [ ] Variables are documented in the README.md (or README.tpl in some charts)
- [ ] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [ ] All test files are added in the tests folder of their respective chart and have a "_test" suffix

### CODE REVIEW

File: charts/shield/templates/host/_configmap_helpers.tpl [hunk 1]
```diff
@@ -100,7 +100,9 @@ true
 {{- $config := dict
   "k8s_cluster_name" .Values.cluster_config.name
   "collector" (include "common.collector_endpoint" .)
-  "collector_port" .Values.sysdig_endpoint.collector.port }}
+  "collector_port" .Values.sysdig_endpoint.collector.port
+  "k8s_delegated_nodes" (dig "k8s_delegated_nodes" 0 .Values.host.additional_settings)
```
Comment (aroberts87) [inline charts/shield/templates/host/_configmap_helpers.tpl]: I imagine we'd want to have a check to see if `features.kubernetes_metadata.enabled` has been set to `false` here. If this feature is turned off I think we'd like the agent/host shield to run with the default values (unless manually overridden of course).

Comment (AlbertoBarba) [reply charts/shield/templates/host/_configmap_helpers.tpl]: @mavimo @irozzo-1A WDYT?

Comment (irozzo-1A) [reply charts/shield/templates/host/_configmap_helpers.tpl]: Yep, I agree that would be better to use the default if `kubernetes_metadata` is disabled

Comment (AlbertoBarba) [reply charts/shield/templates/host/_configmap_helpers.tpl]: Ok, i did apply the changes. User specified configuration will alway rule but, as a default, if the feature is enabled we do force the delegated agents to 0 and, if not, we simply don't set that configuration.

**Review by aroberts87 — APPROVED** [review]

**Pull request merged**